src/app/pages/candidate-details/__tests__/candidate-details.page.spec.ts

```diff
@@ -73,6 +75,7 @@ describe('CandidateDetailsPage', () => {
       providers: [
         { provide: ModalController, useClass: ModalControllerMock },
         { provide: NavParams, useValue: mockNavParams },
+        { provide: Router, useValue: routerSpy },
```

Comment:
We could useClass: RouterMock from ```'@mocks/angular-mocks/router-mock'```

And add a url property to the class i.e.
```
export class RouterMock {
 url = '/url';
.....
}
```

This would then eliminate the need for a spy